### PR TITLE
dat.py - 

### PR DESCRIPTION
**Pull request by daisy-jacobs**
Unsupported unit class + associated changes to read/wrote etc
Next and prev dat functions
Helper functions: _next_in_dat_struct, _prev_in_dat_struct, _ds_label_match, _name_label_match
Small changes to files where global imports used 
Associated release notes and user guide

### CODE REVIEW

File: floodmodeller_api/_base.py [hunk 1]
```diff
@@ -35,7 +35,8 @@ def __init__(self):
             self._filepath = Path(self._filepath).resolve()  # save filepath to class
             # Check if filepath valid
             # * Add check or fix for path lengths greater than DOS standard length of 260 characters
-
+            self.unsupported = dict()
```
Comment (joe-pierce) [inline floodmodeller_api/_base.py:38]: This shouldn't be here, removing

**Comment by joe-pierce** [comment]
Thanks for this Daisy, I've made some changes and pushed to the branch:
- Merged in the latest updates from main code (incl. new test module)
- Removed `self.unsupported = dict()` from FMFile init as this shouldn't be here
- Cleaned up the unsupported unit code - I've also changed the names in the ._unsupported dictionary to include the unit type to avoid any potential name clashes where the unit would be overwritten
- Cleaned up the .next code and added some extra cases

I still need to look at the .prev function, but will have to wait until next week

**Comment by joe-pierce** [comment]
@matthewjwhittle-jacobs wondering if you might know why it's saying the automated tests are failing for this branch. I can confirm all the tests are passing when I run locally. The log suggests it's due to `floodmodeller_api.units.unsupported` not existing which does exist on this branch, but doesn't yet exist on main. Is it something to do with this PR being from a fork? Or is the error happening on test collection?

**Comment by joe-pierce** [comment]
I've made a bit of a restructure to `.prev()` to better capture the complexities of different cases.

**Comment by matthewjwhittle-jacobs** [comment]
Hey! The tests are failling for me locally. I think it is because it tries to import `floodmodeller_api.units.unsupported` but the file is called `Unsupported` (capital U). changing this to lowercase results in tests passing for me! 😄 

I've pushed the change, had to run `git config core.ignorecase false` locally for git to pick it up.

Here is the error I got for ref: 
```
________________________________ ERROR collecting test/test_dat.py ________________________________ 
ImportError while importing test module 'C:\Users\MW055122\Data Science\floodmodeller-api\test\test_dat.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
..\..\.conda\envs\fm-api\Lib\importlib\__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test\test_dat.py:1: in <module>
    from floodmodeller_api import DAT
floodmodeller_api\__init__.py:4: in <module>
    from .zzn import ZZN
floodmodeller_api\zzn.py:24: in <module>
    from ._base import FMFile
floodmodeller_api\_base.py:22: in <module>
    from .units._base import Unit
floodmodeller_api\units\__init__.py:1: in <module>
    from .units import *
floodmodeller_api\units\units.py:26: in <module>
    from .unsupported import UNSUPPORTED
E   ModuleNotFoundError: No module named 'floodmodeller_api.units.unsupported'
```

**Pull request merged**